### ISSUE TITLE
page.ToPDF timeout error doesn't follow global settings

### ISSUE THREAD

**Issue by holytrousers** [Mogztter/asciidoctor-web-pdf#612]
Hello
first of all, i would like to express my gratitude for everybody's work on this project.
```
Unable to generate the PDF - Error: TimeoutError: waiting for Page.printToPDF failed: timeout 30000ms exceeded
> TIP: You can configure the timeout in milliseconds using PUPPETEER_DEFAULT_TIMEOUT, PUPPETEER_NAVIGATION_TIMEOUT or PUPPETEER_RENDERING_TIMEOUT environment variables.
```
This error happens when a certain number of images is inserted into the document.
Setting the appropriate environment variables doesn't help.
Looks like the 30s value is hardcoded in this declaration
```js
const { scale = 1, displayHeaderFooter = false, headerTemplate = '', footerTemplate = '', printBackground = false, landscape = false, pageRanges = '', preferCSSPageSize = false, margin = {}, omitBackground = false, timeout = 30000, } = options;
```   
in `node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js`
hope it helps others with fixing the problem
Cheers !

**Comment by Mogztter** [comment]
We don't specify a timeout when we call [`Page.pdf()`](https://github.com/puppeteer/puppeteer/blob/v13.1.1/docs/api.md#pagepdfoptions).
More than 30 seconds to generate a PDF seems a bit excessive but as you've mentioned maybe Chrome takes a bit of time when the document contains a lot of images.

Anyway, we should declare a new environment variable to configure this timeout: `PUPPETEER_PRINT_TIMEOUT`.

The "default" timeout only applies to navigation (`Page.goto` and `Page.reload`) and rendering (`Page.waitForFunction`).

**Comment by holytrousers** [comment]
> We don't specify a timeout when we call [`Page.pdf()`](https://github.com/puppeteer/puppeteer/blob/v13.1.1/docs/api.md#pagepdfoptions)
Indeed, the problem was with the misleading 'TIP'.
> More than 30 seconds to generate a PDF seems a bit excessive but as you've mentioned maybe Chrome takes a bit of time when the document contains a lot of images.
Actually, the timeout happens when the document exceeds a dozen of pictures ( JPEG files around ~ 3MB size )
These pictures however are inserted in the page layout as running elements positioned inside the margins of the page, which might explain the excessive time.
With the --preview option however, chromium renders the page in a couple of seconds.
> 
> Anyway, we should declare a new environment variable to configure this timeout: `PUPPETEER_PRINT_TIMEOUT`.
> The "default" timeout only applies to navigation (`Page.goto` and `Page.reload`) and rendering (`Page.waitForFunction`).
And a different message for this specific timeout.
It might be even better to create a separate 'TIP' for each timeout since it actually isolates the problem.
Thank you for your help !

**Comment by Mogztter** [comment]
Fixed in 643d5548bc9953e592957ccc356162b8523ab672
